lib/Dialect/Comb/CombFolds.cpp

```diff
@@ -935,6 +1033,16 @@ LogicalResult OrOp::canonicalize(OrOp op, PatternRewriter &rewriter) {
   if (tryFlatteningOperands(op, rewriter))
     return success();
 
+  // or(..., concat(x, cst1), concat(cst2, y)
+  //    ==> or(..., concat(x, cst3, y)), when x and y don't overlap.
+  for (size_t i = 0; i < size - 1; ++i) {
+    if (auto concat = inputs[i].getDefiningOp<ConcatOp>())
+      for (size_t j = i + 1; j < size; ++j)
+        if (auto concat = inputs[j].getDefiningOp<ConcatOp>())
+          if (canonicalizeOrExclusiveConcat(op, i, j, rewriter))
+            return success();
+  }
```

Comment:
nice

Reply:
I can't take credit for this, since I saw this pattern being used for the `canonicalizeLogicalCstWithConcat()` function just a few lines above this, but I agree that it's pretty neat.